### PR TITLE
Implement Single Chapter Downloads

### PR DESCRIPTION
**Pull request by khalidzahra**
This PR first migrates the command-line arguments to be specified using flags leveraging the _flag_ library in golang as I found this to be more appropriate and user-friendly. Moreover, users can download specific chapters by their IDs using the _--chapterid_ flag, and by their names using the _--chapter_ flag.

### CODE REVIEW

File: app.go [hunk 1]
```diff
@@ -1,56 +1,36 @@
...
 )
 
 func main() {
-	locale := "ar"
-	concurrentDownloads := 10
-	reverseOrder := false
-	outputDir, err := os.Getwd()
+	locale := flag.String("locale", "ar", "--locale <language>")
+	concurrentDownloads := flag.Int("batches", 10, "--batches <amount>")
+	reverseOrder := flag.Bool("reverse", false, "--reverse <true/flase>")
```
Comment (LeOndaz) [inline app.go]: typo `true/flase`

**Comment by LeOndaz** [comment]
Thanks for your efforts, I've made some comments over the codebase and this is a general comment:

- We're using pointers everywhere from `flags`, we could force `flags` to inject the value into a specific variable, so that we use them as normal variables and not pointers, explained [here](https://pkg.go.dev/flag)

File: app.go (hunk 1 above)
Comment (khalidzahra) [reply app.go]: Thank you for pointing this out, I've fixed it in my latest commit.

**Comment by khalidzahra** [comment]
I have made the flag changes that you requested, we no longer handle pointers in app.go.

**Pull request merged**